Errors when building on a 32 bit architecture

When building on i686 get the following errors:
```
/builddir/build/BUILD/egaroucid-7.8.1_20260521git9603cd2-build/Egaroucid-9603cd2fb419f53fec33ba1d287b579f3d35ffc4/src/engine/hash.hpp:71:5: error: narrowing conversion of ‘4294967296’ from ‘long long int’ to ‘size_t’ {aka ‘unsigned int’} [-Wnarrowing]
   71 |     4294967296
```

From https://github.com/Nyanyan/Egaroucid/blob/main/src/engine/hash.hpp#L18 maybe conditional compilation should be used?

Full build log at https://copr.fedorainfracloud.org/coprs/fed500/egaroucid/build/10492430/

--- a/src/engine/hash.hpp
+++ b/src/engine/hash.hpp
@@ -15,7 +15,11 @@
 /*
     @brief definition of maximum hash level
 */
-constexpr int N_HASH_LEVEL = 33; // level 32 is maximim because hash is 32 bit
+#ifdef HAS_32_BIT_OS
+constexpr int N_HASH_LEVEL = 32; // level 32 is maximim because hash is 32 bit
+#else
+constexpr int N_HASH_LEVEL = 33;
+#endif
 constexpr int DEFAULT_HASH_LEVEL = 25;
 
 #if USE_CHANGEABLE_HASH_LEVEL
@@ -35,6 +39,43 @@ uint32_t hash_rand_opponent[4][65536];
 
     2 ^ hash_level will be tha size of hash
 */
+
+#ifdef HAS_32_BIT_OS
+constexpr size_t hash_sizes[N_HASH_LEVEL] = {
+    1,
+    2,
+    4,
+    8,
+    16,
+    32,
+    64,
+    128,
+    256,
+    512,
+    1024,
+    2048,
+    4096,
+    8192,
+    16384,
+    32768,
+    65536,
+    131072,
+    262144,
+    524288,
+    1048576,
+    2097152,
+    4194304,
+    8388608,
+    16777216,
+    33554432,
+    67108864,
+    134217728,
+    268435456,
+    536870912,
+    1073741824,
+    2147483648
+};
+#else
 constexpr size_t hash_sizes[N_HASH_LEVEL] = {
     1,
     2,
@@ -70,6 +111,7 @@ constexpr size_t hash_sizes[N_HASH_LEVEL] = {
     2147483648,
     4294967296
 };
+#endif
 
 /*
     @brief initialize hash array randomly

